File: GiftifyBackend/Giftify/src/main/java/it/unicam/cs/Giftify/Model/WishList.java
package it.unicam.cs.Giftify.Model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
public class WishList {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @OneToMany
    @Getter
    @Setter
    private List<Wish> wishes;
    @ManyToOne
    @Getter
    private User user;

    public WishList(User user) {
        wishes= new ArrayList<>();
        this.user = user;
    }

    public WishList() {
    }

}

File: GiftifyFrontend/Giftify/src/model/WishList.ts
import { Wish } from "./Wish";
import {Account} from "./Account";
import {Community} from './Community';

export class WishList {
  id: number;
  wishes: Wish[];
  user: Account;

  constructor(id: number, user: Account) {
    this.id = id;
    this.wishes = [];
    this.user = user;
  }

}
